### PR TITLE
[flappy-bird] feat: enhancements

### PR DESCRIPTION
**Pull request by jeffsieu**
- Adds FlappyBirdCharacterTag component with `characterId`. Replaces the `characterIdToEntityMap`
- When bird contacts wall, `FlappyBirdWallHitEvent` is emitted.
- On `FlappyBirdWallHitEvent`, bird that hit wall is removed

### CODE REVIEW

File: ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift [hunk 1]
```diff
@@ -30,7 +30,8 @@ enum FlappyBirdEntityCreator {
                              impulse: impulseValue,
                              categoryBitMask: FlappyBirdPhysicsCategory.character,
                              collisionBitMask: FlappyBirdPhysicsCategory.wall | FlappyBirdPhysicsCategory.ceiling,
```
Comment (chownces) [inline ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift:32]: Let's use contactTestBitMask only, against just the `.wall`. Anw we only register the collisionHandler between .wall and .character

File: ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift [hunk 2]
```diff
@@ -156,7 +158,7 @@ extension FlappyBirdEntityCreator {
                              isDynamic: isDynamic,
                              categoryBitMask: physicsType,
                              collisionBitMask: FlappyBirdPhysicsCategory.character,
```
Comment (chownces) [inline ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift:160]: Same as above

**Review by chownces — APPROVED** [review]

File: ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift [hunk 3]
```diff
@@ -29,8 +29,9 @@ enum FlappyBirdEntityCreator {
                              linearDamping: 10,
                              impulse: impulseValue,
                              categoryBitMask: FlappyBirdPhysicsCategory.character,
-                             collisionBitMask: FlappyBirdPhysicsCategory.wall | FlappyBirdPhysicsCategory.ceiling,
-                             contactTestBitMask: FlappyBirdPhysicsCategory.none)
+                             collisionBitMask: FlappyBirdPhysicsCategory.none,
+                             contactTestBitMask: FlappyBirdPhysicsCategory.wall | FlappyBirdPhysicsCategory.ceiling | FlappyBirdPhysicsCategory.scoringArea),
```
Comment (chownces) [inline ArkGameExample/games/FlappyBird/FlappyBirdEntityCreator.swift:33]: I think can remove the bitmask for ceiling. Although its emitted, the contact event for ceiling is unused

**Review by markusyeo — APPROVED** [review]
LGTM

**Pull request merged**